src/main/java/Parser.java

```diff
@@ -0,0 +1,37 @@
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.regex.Pattern;
+
+/**
+ * Deals with the parsing of LocalDateTime objects to and from strings.
+ */
+public class Parser {
```

Comment:
Might want to add `private Parser() {}` to ensure no instance of `Parser` can be created.